### PR TITLE
Re-write sequence component

### PR DESCRIPTION
**Pull request by wesleyyee**
## Hey, I just made a Pull Request!

Material UI provides ways to create both horizontal and vertical steppers. Our internal stepper component, currently supports both but in doing so, adds complexity since they are composed differently. This complexity led us to implement the steps awkwardly (In an object with a property containing some metadata about the step as well as a property that contained the jsx of the inner content). 

After looking at the `Stepper` usage internally we *only* have vertical steppers with the exception of skill exchange which is still a WIP. My proposal is to only support vertical in our custom implementation to provide a more declarative and simpler experience for our users. In addition, it also greatly simplifies using forms inside a stepper and was the initial reason i wanted to look into refactoring this.

Note: users can still build horizontal steppers with material ui, but we won't support it in our easy to use stepper. 

_This PR also renames `Sequence` to `Stepper` which is the updated name in material._

**Before**:
Screen Shot 2020-04-29 at 11 04 35 PM

**After**:
Screen Shot 2020-04-29 at 11 02 17 PM

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by freben** [comment]
I've been thinking about naming.

It's sometimes been confusing that our internal components were named the exact same as MUI components. At the same time, it makes it easier to make the link to what underlying component they represent. Let's talk prefixes?

I could envision at least naming this `VerticalStepper` and `Vertical[Stepper]Step` to begin with. That opens the possibility to make a ticket (or write ourselves) the horizontal cousin of it later.

But then, what if we actually started prefixing our simplified wrappers with `Simple` or `Basic`? What do you think?

File: packages/core/src/components/Sequence/Step.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+const Step: FC<StepProps> = ({
+  title,
+  children,
+  end,
+  stepperLength,
+  stepIndex,
+  setStepIndex,
+  stepArray,
+  setStepArray,
```
Comment (freben) [inline packages/core/src/components/Sequence/Step.tsx:55]: I'm not entirely clear on why the step array exists explicitly and can be set to a new value.

File: packages/core/src/components/Sequence/Step.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  actions?: StepActions;
+};
+
+const Step: FC<StepProps> = ({
+  title,
+  children,
+  end,
+  stepperLength,
+  stepIndex,
+  setStepIndex,
```
Comment (freben) [inline packages/core/src/components/Sequence/Step.tsx:53]: The three items above may be seen as power features perhaps? I don't mind much tho

Comment (freben) [reply packages/core/src/components/Sequence/Step.tsx:53]: Oh now I think I see - these are part of the public API of the Step even though the user is never supposed to set them, is that right?

Could you wrap the Stepper in a react context instead, that manages the state and actions handling what the current step is etc? The user should only be exposed to the exact props that they can affect.

File: packages/core/src/components/Sequence/Step.tsx (hunk 1 above)
Comment (wesleyyee) [reply packages/core/src/components/Sequence/Step.tsx:55]: so, right now we have steppers that can have a next button that can skip slides depending on answers. so we need to keep a history of the steps for the back button. i should rename this `stepHistory` instead

**Comment by wesleyyee** [comment]
@freben thanks for taking a look! yea, this PR definitely needs to be cleaned up but wanted to get thoughts on the API change

File: packages/core/src/components/Sequence/Step.tsx (hunk 2 above)
Comment (wesleyyee) [reply packages/core/src/components/Sequence/Step.tsx:53]: yea, they're not exposed to this. The prop is managed in `Stepper` and cloned into its children (`Step`)

Comment (wesleyyee) [reply packages/core/src/components/Sequence/Step.tsx:53]: i'll break apart the prop typing to make this clear, cause you're right its not clear at all

Comment (wesleyyee) [reply packages/core/src/components/Sequence/Step.tsx:53]: also document it. but the user facing props are title, children and actions

Comment (wesleyyee) [reply packages/core/src/components/Sequence/Step.tsx]: ah yeap context makes much more sense @freben, my brain wasn't working

**Comment by wesleyyee** [comment]
@freben should be updated with your feedback!

**Comment by wesleyyee** [comment]
@freben @Rugvip bump :)

**Review by Rugvip — APPROVED** [review]
👍 to removing UX choices that didn't really provide value :grin:

Not crazy about the naming xP `BasicVerticalStepper` is a bit long, and I think there's a risk you'd confuse `Basic*` with something that is lower-level. I think the `Simple*` prefix might be a bit more tempting :grin: So `SimpleStepper`, `SimpleStepperStep` etc.

**Comment by wesleyyee** [comment]
@Rugvip 👍 yea agree the name is long. so you'd vote for also removing the `Vertical` as well

**Comment by wesleyyee** [comment]
yea i'm cool with `Simple` is that ok with you @freben ?

**Comment by Rugvip** [comment]
@wesleyyee I'd vote for removing `Vertical` with the assumption that vertical will be the main choice, and if we see a good reason to add a horizontal one in the future that would still be a secondary one, likely called `SimpleHorizontalStepper` or just `HorizontalStepper`.

**Comment by freben** [comment]
Fine with Simple :)

Sure remove Vertical if y'all want to, but I don't think we should be afraid of identifier length particularly for something that will be very rarely used :) I'd expect that there will be a horizontal one later, but can't know of course

**Comment by wesleyyee** [comment]
@freben you don't think users will use this component?

**Review by Rugvip — APPROVED** [review]

**Comment by freben** [comment]
No no - i definitely think it'll be used, I meant just the word in the component name :)

**Pull request merged**